src/Slot-Examples/LazySlot.class.st

```diff
@@ -26,12 +26,14 @@ LazySlot >> emitValue: aMethodBuilder [
 		pushLiteral: nil;
 		send: #==;
 		jumpAheadTo: #target if: false;
-		popTop;
-		pushLiteral: default.
-	isBlock ifTrue: [ 
+		popTop.
+	default isBlock 
```

Comment:
Having so many "default isBlock ifTrue/ifFalse" statements it looks like we need to introduce object here. Like VariableInitializer with subclasses ConstantVariableInitializer and BlockVariableInitializer. And replace the #default var with #initialzer var.

It could also allow to avoid some subclasses of ClassVariable. I wonder if we can move #initializer variable directly to the base class ClassVariable which will be NullVariableInitializer by default.

Reply:
I will think about that